src/app/results/results.component.ts

```diff
@@ -184,6 +185,12 @@ export class ResultsComponent implements OnInit {
       this.begin = this.start + 1;
 
     });
+
+    if (this.searchdata.query === '') {
```

Comment:
these lines are executed only once, but what we need is to get executed each time query is changed.
add these lines into the above subscribe block of `wholequery$` at 183 line